monte_carlo.py

```diff
@@ -23,60 +27,229 @@
+            #test ymin
+            r = ymind/(sin(theta)*sin(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('y_min')
+            elif r == tempd:
+                lim_bound.append('y_min')               
+            
+            #test zmax
+            r = zmaxd/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_max')
+            elif r == tempd:
+                lim_bound.append('z_max')              
+                
+            #test zmin
+            r = zmind/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_min')
+            elif r == tempd:
+                lim_bound.append('z_min')               
+                
+            # if the neutron stays within the bounding box
+            if lim_bound == []:
+                neutron_distance = 0
+            neutron_position.sadd(neutron_movement_unit_vector * tempd)
+
+            for surface in lim_bound:  
+                # if the neutron is reflected
+                if (bounds.get_surface_type(surface) == 1):
+                    if surface == 'x_max' or surface == 'x_min':
+                        neutron_movement_unit_vector.smul([-1, 1, 1])
+                    if surface == 'y_max' or surface == 'y_min':
+                        neutron_movement_unit_vector.smul([1, -1, 1])
+                    if surface == 'z_max' or surface == 'z_min':
+                        neutron_movement_unit_vector.smul([1, 1, -1])
+                    neutron_reflected = True
+    
+                # if the neutron escapes
+                elif (bounds.get_surface_type(surface) == 0):
+                    neutron_lost = True
+                    neutron_interaction = 1
+                    neutron_distance = 0
+                    tallies["leaks"].increment(1)
+            if neutron_reflected:
+                neutron_distance -= tempd
+                
+        if not neutron_lost:
+            neutron_interaction = sample_interaction(mat)
 
+    # neutron_position turns into a vector pointing from the neutron's origin
+    # to its absorption point
+    if not neutron_lost:
+        tallies["absorptions"].increment(1)
+        if sample_fission(mat) == 1:
+            for j in xrange(sample_num_fission(mat)):
+                fission_banks["new"].add([neutron_position.x,
+                        neutron_position.y, neutron_position.z])
+                tallies["fissions"].increment(1) 
+        neutron_position.ssub([neutron_starting_point.x,
+            neutron_starting_point.y, neutron_starting_point.z])
```

Comment:
Using -= and += on Coords objects caused errors to happen when I later tried to use Coords functions on those objects. It seemed like using 
neutron_position -= neutron_starting_point 
turned the neutron_position into a list or some other data type, because later when I tried to use neutron_position.getDistance(), I got an error.